src/ng/httpBackend.js

```diff
@@ -88,6 +88,20 @@ function createHttpBackend($browser, createXhr, $browserDefer, callbacks, rawDoc
       xhr.onerror = requestError;
       xhr.onabort = requestError;
 
+      if (eventHandlers) {
+        forEach(eventHandlers, function(value, key) {
+          if (key !== 'upload') {
+            xhr.addEventListener(key, value);
+          }
+        });
+
+        if (eventHandlers.upload) {
```

Comment:
Sorry, this is a typo on my part. This is meant to attach events on `xhr.upload`. The line 100 should read `xhr.upload.addEventListener(key, value);`.